### PR TITLE
[Calendar] handle format exception when parse date time

### PR DESCRIPTION
**Pull request by KayMKM**
## Description

Handle format exception when parse date time

## Testing Steps

In create meeting flow, when bot asked for date or time, say "an hour".

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the appropriate unit tests
- [ ] I have updated related documentation

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
- [ ] A duplicate issue is filed to track future  work.

If you have updated responses or `.lu` files:
- [ ] All languages have been updated
- [ ] You have tested deployment with your new models

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs [hunk 1]
```diff
@@ -713,29 +713,35 @@ public async Task<DialogTurnResult> AfterUpdateStartDateForCreate(WaterfallStepC
...
+                                    }
+
+                                    state.StartDate.Add(isRelativeTime ? TimeZoneInfo.ConvertTime(dateTime, TimeZoneInfo.Local, state.GetUserTimeZone()) : dateTime);
                                 }
-
-                                state.StartDate.Add(isRelativeTime ? TimeZoneInfo.ConvertTime(dateTime, TimeZoneInfo.Local, state.GetUserTimeZone()) : dateTime);
+                            }
+                            catch (FormatException)
+                            {
                             }
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs:747]: we should at least log the exception and add trace, if we decide not to surface the error to the user

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs (hunk 1 above)
Comment (KayMKM) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs:747]: fixed

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs [hunk 2]
```diff
@@ -798,12 +806,18 @@ public async Task<DialogTurnResult> AfterUpdateStartTimeForCreate(WaterfallStepC
...
 
-                            if (dateTime != null)
+                                if (dateTime != null)
+                                {
+                                    var isRelativeTime = IsRelativeTime(sc.Context.Activity.Text, dateTimeValue, dateTimeConvertType);
+                                    state.StartTime.Add(isRelativeTime ? TimeZoneInfo.ConvertTime(dateTime, TimeZoneInfo.Local, state.GetUserTimeZone()) : dateTime);
+                                }
+                            }
+                            catch (FormatException)
                             {
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs:820]: log the exception here as well

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs (hunk 2 above)
Comment (KayMKM) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Dialogs/CreateEvent/CreateEventDialog.cs:820]: fixed

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**